#! /bin/sh -e

# All lines beginning with `# DP:' are a description of the patch.
# DP: Description: Only parse three parts of a kernel version number.
# DP: Related bugs: 218546
# DP: Author: Daniel Jacobowitz <dan@debian.org>
# DP: Upstream status: Debian-specic
# DP: Status details: Rejected by Ulrich, needed for Debian.
# DP:       http://sources.redhat.com/ml/libc-alpha/2003-11/msg00025.html
# DP: Date: 2003-11-01

if [ $# -ne 2 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -d "$2" -f --no-backup-if-mismatch -p1 < $0;;
    -unpatch) patch -d "$2" -f --no-backup-if-mismatch -R -p1 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

# append the patch here and adjust the -p? flag in the patch calls.
2003-11-01  Daniel Jacobowitz  <dan@debian.org>

	* sysdeps/unix/sysv/linux/dl-osinfo.h (DL_SYSDEP_OSCHECK): Don't
	parse more than three parts of the version number.

--- glibc-2.3.2/sysdeps/unix/sysv/linux/dl-osinfo.h.orig	2003-11-01 14:20:45.000000000 -0500
+++ glibc-2.3.2/sysdeps/unix/sysv/linux/dl-osinfo.h	2003-11-01 14:21:32.000000000 -0500
@@ -91,7 +91,7 @@
 	    version <<= 8;						      \
 	    version |= here;						      \
 									      \
-	    if (*cp++ != '.')						      \
+	    if (*cp++ != '.' || parts == 3)				      \
 	      /* Another part following?  */				      \
 	      break;							      \
 	  }								      \
